server/consumer_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2016-2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
```

Comment:
There already is a file with consumer tests: `jetstream_consumer_test.go`

Reply:
From a quick look at the file I had the impression that the tests living there are rather integration tests and not unit tests. Would you like me to move the newly added test code there?

Reply:
The existing files are easier since they all tie into our parallel CI/CD.

Reply:
Okay, I have pushed a commit on the PR that moves the newly added test and benchmark to the existing file.